extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/knative/events/CloudEventBuilder.java

```diff
@@ -0,0 +1,195 @@
+package io.quarkus.funqy.knative.events;
+
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import java.util.Map;
+
+public class CloudEventBuilder {
+    private CloudEvent.SpecVersion specVersion;
+    private String id;
+    private String type;
+    private String source;
+    private String subject;
+    private OffsetDateTime time;
+    private String dataSchema;
+    private Map<String, String> extensions;
+
+    private CloudEventBuilder() {
+    }
+
+    public static CloudEventBuilder create() {
+        return new CloudEventBuilder();
+    }
+
+    public CloudEventBuilder specVersion(CloudEvent.SpecVersion specVersion) {
```

Comment:
@patriot1burke I could remove `specVersion()` and `create()` here and introduce `createV1()` and `createV03()`?

Reply:
Or maybe not, this would disallow automatic defaulting.